ond/services/chat/stream.go

```diff
@@ -85,24 +85,56 @@ func (s *StreamConsumer) Consume() {
 				Done:  true,
 			}
 			break
-		}
-
-		parsedString := strings.Split(line, "data:")
-		jsonString := parsedString[1]
+		} else if strings.HasPrefix(line, "data:[ERROR]:") {
+			parsedString := strings.Split(line, "data:[ERROR]:")
+			jsonString := parsedString[1]
+
+			var errResp errors.ErrResponse
+			if err := json.Unmarshal([]byte(jsonString), &errResp); err != nil {
+				s.EventChan <- Event{
+					Error: &errors.ErrResponse{
+						Message:   err.Error(),
+						ErrorCode: errors.ErrAPIClientError.String(),
+						Status:    0,
+					},
+				}
+				break
+			}
 
-		if err := json.Unmarshal([]byte(jsonString), &eventData); err != nil {
-			s.EventChan <- Event{Error: err}
+			s.EventChan <- Event{
+				Error: &errResp,
+			}
 			break
-		}
+		} else {
```

Comment:
probably dont need this `else` block. We can instead move the code outside of the if-else blocks and have it the way it already is.

Reply:
Fixed